R/zchunk_LB123.LC_R_MgdPastFor_Yh_GLU.R

```diff
@@ -35,102 +35,314 @@ module_aglu_LB123.LC_R_MgdPastFor_Yh_GLU_DISABLED <- function(command, ...) {
     L121.CarbonContent_kgm2_R_LT_GLU <- get_data(all_data, "L121.CarbonContent_kgm2_R_LT_GLU")
     L121.Yield_kgm2_R_Past_GLU <- get_data(all_data, "L121.Yield_kgm2_R_Past_GLU")
     L101.Pop_thous_R_Yh <- get_data(all_data, "L101.Pop_thous_R_Yh")
 
-    # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
+    # Part 1: Pasture production, yield and managed pasture land
+
+    # Calculate bottom-up estimates of total pasture grass production by region and GLU.
+    # Start with total pasture land data
+    L120.LC_bm2_R_LT_Yh_GLU %>%
+      filter(Land_Type == "Pasture", year %in% AGLU_HISTORICAL_YEARS) ->
+      L123.LC_bm2_R_Past_Y_GLU
+
+    # Match total pasture land area and yield data to calculate potential production
+    L123.LC_bm2_R_Past_Y_GLU %>%
+      left_join_error_no_match(L121.Yield_kgm2_R_Past_GLU, by = c("GCAM_region_ID", "Land_Type", "GLU")) %>%
+      # Calculate potential production as total land times yield
+      mutate(value = value * pasture_yield) ->
+      L123.ag_potentialProd_Mt_R_Past_Y_GLU
+
+    # Use this "potential production" by region and GLU to disaggregate actual pasture land production to GLUs.
+    # Assume regional consumption equals managed pasture land production.
+    L123.ag_potentialProd_Mt_R_Past_Y_GLU %>%
+      # Calculate regional total potential production
+      group_by(GCAM_region_ID, Land_Type, year) %>%
+      summarise(total = sum(value)) %>%
+      ungroup %>%
+      # Match in potential production by region and GLU to calculate the fraction
+      right_join(L123.ag_potentialProd_Mt_R_Past_Y_GLU, by = c("GCAM_region_ID", "Land_Type", "year")) %>%
+      # Calculate the GLU to region fraction
+      mutate(frac = value / total) %>%
+      select(-total, -value, -pasture_yield) %>%
+      # Drop NAs for GLUs/years with no managed pasture
+      replace_na(list(frac = 0)) %>%
+      rename(GCAM_commodity = Land_Type) %>%
+      # Match in regional pasture consumption (managed pasture land production) for disaggregation
+      # Note: left_join_error_no_match fails timeshift, due to NAs
```

Comment:
Here, `left_join_error_no_match` fails under timeshift due to NAs...tried figuring out why, but not exactly sure...